src/snooze/ecs/minigame/vermins/VerminsMiniGameSystem.cpp

```diff
@@ -110,12 +117,34 @@ void VerminsMiniGameSystem::OnMiniGameStart()
         forge::builtin::EntityClickedEvent::Handler(this, &VerminsMiniGameSystem::OnEntityClickedEvent);
 
     static std::random_device rd;
-    m_FirstRun = true;
-    m_MaxVermins = rd() % 11;
-    m_Timer.Start(m_TimeToVermin);
-    // TODO: Random (or predefined) tool location
+    m_MaxVermins = rd() % 5 + 1;
+
+    m_Lock = forge::DataAPI::GetDataFrom<EntityCatalog>(DataList::Entity::LockLock);
+    m_Lock->SetPosition(50.f - m_Lock->GetSize().w * 0.5f, 50.f - m_Lock->GetSize().d * 0.5f, 4.f);
+    RequestAddEntity(m_Lock);
+    m_MetalBox = forge::DataAPI::GetDataFrom<EntityCatalog>(DataList::Entity::LockMetalBox);
+    m_MetalBox->SetPosition(50.f - m_MetalBox->GetSize().w * 0.5f, 50.f - m_MetalBox->GetSize().d * 0.5f, 3.f);
+    RequestAddEntity(m_MetalBox);
+
+    forge::Vector<forge::Vector3f> spawnPoints;
+    for (u32 i = 0; i < SnoozeConfig::MobSpawnPointCount; i++)
+    {
+        spawnPoints.push_back(SnoozeConfig::MobSpawnPoints[i]);
+    }
+    std::random_shuffle(spawnPoints.begin(), spawnPoints.end());
+
+    // Populating the array with vermins
+    for (int i = 0; i < m_MaxVermins; i++)
+    {
+        forge::Entity::Ptr vermin = forge::DataAPI::GetDataFrom<EntityCatalog>(DataList::Entity::VerminEnemy);
+        forge::Vector3 randLocation = spawnPoints[i];
+        vermin->SetPosition(randLocation.x, randLocation.y, randLocation.z);
```

Comment:
vermin->SetPosition(spawnPoints[i].x, spawnPoints[i].y, spawnPoints[i].z) would result in one less allocation on the stack :)